ui/v2/src/components/Settings/SettingsAboutPanel.tsx

```diff
@@ -26,32 +28,72 @@ export const SettingsAboutPanel: FunctionComponent<IProps> = (props: IProps) =>
     );
   }
 
-  function renderVersion() {
-    if (!data || !data.version) { return; }
+  function maybeRenderLatestVersion() {
+    if (!dt || !dt.latestversion || !dt.latestversion.shorthash || !dt.latestversion.url) { return; }
+    if (!data || !data.version || !data.version.hash) {
+      return (
+        <>{dt.latestversion.shorthash}</>
+      );
+    }
+
+    if (data.version.hash !== dt.latestversion.shorthash) {
+      return (
+        <>
+          <strong>{dt.latestversion.shorthash} [NEW] </strong><a href={dt.latestversion.url}>Download</a>
+        </>
+      );
+    }
+
+    return (
+      <>{dt.latestversion.shorthash}</>
+    );
+  }
+
+  function RenderLatestVersion() {
     return (
-      <>
       <HTMLTable>
         <tbody>
-          {maybeRenderTag()}
           <tr>
-            <td>Build hash:</td>
-            <td>{data.version.hash}</td>
+            < td>Latest Version Build Hash: </td>
+            < td>{maybeRenderLatestVersion()} </td>
           </tr>
           <tr>
-            <td>Build time:</td>
-            <td>{data.version.build_time}</td>
+            <td><Button minimal={true} onClick={() => refetch()} text="Check for new version!" /></td>
           </tr>
-        </tbody>  
+        </tbody>
       </HTMLTable>
+    );
+  }
+
+  function renderVersion() {
+    if (!data || !data.version) { return; }
+    return (
+      <>
+        <HTMLTable>
+          <tbody>
+            {maybeRenderTag()}
+            <tr>
+              <td>Build hash:</td>
+              <td>{data.version.hash}</td>
+            </tr>
+            <tr>
+              <td>Build time:</td>
+              <td>{data.version.build_time}</td>
+            </tr>
+          </tbody>
+        </HTMLTable>
       </>
     );
   }
   return (
     <>
       <H4>About</H4>
       {!data || loading ? <Spinner size={Spinner.SIZE_LARGE} /> : undefined}
-      {!!error ? <span>error.message</span> : undefined}
+      {!!error ? <span>{error.message}</span> : undefined}
+      {!!er ? <span>{er.message}</span> : undefined}
       {renderVersion()}
+      {!dt || ld || networkStatus == 4 ? <Spinner size={Spinner.SIZE_SMALL} /> : undefined}
+      {RenderLatestVersion()}
```

Comment:
I think these variables (`dt`, `ld`, `er`) need to be better named.

I would make it so that if the spinner is showing (ie the query is in progress) then `renderLatestVersion` is not called.

Reply:
should be covered with latest commit ....